### PR TITLE
test: add 100% test coverage for `array_list`, `hex_color`, `hostname`, `json` and `range` validators

### PR DESCRIPTION
**Pull request by Biplab-Dutta**

### CODE REVIEW

File: lib/src/validators/hex_color.dart [hunk 1]
```diff
@@ -20,7 +20,7 @@ class HexColor extends Validator {
   @override
   bool isValid(dynamic value) {
     if (value is String &&
-        RegExp('/^([A-Fa-f0-9]{6}|[A-Fa-f0-9]{3})\$/').hasMatch(value)) {
```
Comment (lohanidamodar) [inline lib/src/validators/hex_color.dart:23]: we want the color value without the `#`.

File: test/unit/validators/hex_color_test.dart [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+          expect(hexColor.isArray(), false);
+        },
+      );
+
+      test(
+        'isValid(): should return true if the string is a valid string',
+        () {
+          expect(hexColor.isValid(1), false);
+          expect(hexColor.isValid('abc'), false);
+          expect(hexColor.isValid('#FF7723'), true);
```
Comment (lohanidamodar) [inline test/unit/validators/hex_color_test.dart:38]: this should be false, and `ff7723` should be true.

File: test/unit/validators/hex_color_test.dart [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+        () {
+          expect(hexColor.isArray(), false);
+        },
+      );
+
+      test(
+        'isValid(): should return true if the string is a valid string',
+        () {
+          expect(hexColor.isValid(1), false);
+          expect(hexColor.isValid('abc'), false);
```
Comment (lohanidamodar) [inline test/unit/validators/hex_color_test.dart:37]: This should be true

Comment (lohanidamodar) [reply test/unit/validators/hex_color_test.dart:37]: Let's use this as reference to test
https://github.com/utopia-php/framework/blob/master/tests/Validator/HexColorTest.php

File: test/unit/validators/hostname_test.dart [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+        'isArray(): should return false',
+        () {
+          expect(hostname.isArray(), false);
+        },
+      );
+
+      test(
+        'isValid(): should check the validity and return proper boolean value',
+        () {
+          expect(hostname.isValid(1), false);
```
Comment (lohanidamodar) [inline test/unit/validators/hostname_test.dart:37]: Let's add more tests that tests the wildcard and all. 
ref: https://github.com/utopia-php/framework/blob/master/tests/Validator/HostnameTest.php

File: test/unit/validators/array_list_test.dart [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+import 'package:utopia_framework/utopia_validators.dart';
+
+import 'mock_validator.dart';
+
+String _description(String value) {
+  return 'Value must be a valid array and $value';
+}
+
+void main() {
+  final mockValidator = MockValidator();
```
Comment (lohanidamodar) [inline test/unit/validators/array_list_test.dart:11]: I know it makes this dependent to other validators, but let's instead test this with real Text, and other validators we already have ?

**Review by lohanidamodar — CHANGES_REQUESTED** [review]
Hey, thank you for your great work, just few comments to make it robust. Take more ref from
https://github.com/utopia-php/framework/tree/master/tests/Validator

File: test/unit/validators/array_list_test.dart (hunk 5 above)
Comment (Biplab-Dutta) [reply test/unit/validators/array_list_test.dart:11]: But won't that be against the ethics of unit testing? Passing real dependencies in test? Also, I thought people can learn how to create Mocks on their own. So, I on purpose created it manually without using Mocktail. What do you think?

Comment (lohanidamodar) [reply test/unit/validators/array_list_test.dart:11]: Okay thats all right, let's later add one in e2e test that tests ArrayList validator with real validators.

**Review by lohanidamodar — APPROVED** [review]

**Pull request merged**